applications/FluidDynamicsApplication/custom_utilities/mass_conservation_utility.cpp

```diff
@@ -30,131 +29,211 @@ namespace Kratos
 /* Public functions *******************************************************/
 
 /// constructor
-MassConservationCheckProcess::MassConservationCheckProcess(
-        ModelPart& rModelPart,
-        const bool PerformCorrections,
-        const int CorrectionFreq,
-        const bool WriteToLogFile,
-        const std::string LogFileName)
-    : Process(), mrModelPart(rModelPart) {
-
-    mCorrectionFreq = CorrectionFreq;
-    mWriteToLogFile = WriteToLogFile;
-    mLogFileName = LogFileName;
-    mPerformCorrections = PerformCorrections;
+MassConservationUtility::MassConservationUtility(
+    Model& rModel,
+    Parameters Settings)
+    : mrModelPart(rModel.GetModelPart(Settings["model_part_name"].GetString())) {
```

Comment:
We are retrieving the `model_part_name` from the json before having checked the defaults.

Reply:
mrModelPart is a reference so it has to be initialized here...